index.rst

```diff
@@ -592,6 +594,99 @@ Parsing XML Files
+Furthermore, it is rare that more than one client issues commands to the same CSC and, in such cases, delivery order is probably not going to be critical.
+In any case, it is not advisable to partition command topics.
+
+For events and telemetry topics, the situation is a bit more nuanced.
+
+Event topics, on the other hand, are asynchronously published by components to indicate state changes.
+Our system relies heavily on event topic delivery and even more so on ordered delivery.
+As such, we will not support partitioning of events.
+
+Telemetry is issued at a fixed rate, that can go up to 50Hz per system requirements.
+Since telemetry is published at a fixed rate, out of order delivery (especially of high throughput topics) is not really a problem, so long as the application layer discards older topics.
+In these situations, an out of order delivery would be treated as a missed delivery, which our system is designed to be resilient to.
+Finally, telemetry topics seems to be the least affected by topic partitioning and, as such, we provide support for partitioning of telemetry topics.
+However, it is advisable to use topic partitioning only in particular high throughput cases where it might be necessary.
+
+On the application layer we enforce these criterias by applying the following rules;
+
+- All topics are created with 1 partition.
+- Command and event topics are written with a provided key.
+  Kafka ensures that topics with the same key value are all written to a single partition.
+  Therefore, even if someone inadvertently increases the number of partitions for a command or event topic, we still get the single partition behavior.
+- Telemetry topics are written with no key.
+  This allows the operators to increase the number of partitions for telemetry topics and, Kafka will automatically handle writing samples to different partitions.
+
+In order to access the impact of increasing partition number in our system, we executed a read/write test for different publishing rates and measured the number of out of order topic delivery.
+For this test we used the same setup as the Performance tests, using the M1M3 ``forceActuatorData`` topic, which is the largest topic in the system.
+The topic partition was increased to 32 (which is the number of partitions used by the sasquatch connector :sqr:`068`) using the ``kafka-topics`` cli tool shipped with Kafka tools.
+We did not experimented with different partitioning values as the number is well established at this point.
+
+The command used to expand the number of partitions was:
+
+.. prompt:: bash
+
+   kafka-topics --bootstrap-server $LSST_KAFKA_BROKER_ADDR --topic "lsst.${LSST_TOPIC_SUBNAME}.MTM1M3.forceActuatorData" --alter --partitions 32
+
+The results are shown in the table and figure below.
+
++--------------+--------------+
+| Publish rate | message loss |
```

Comment:
You refer to `message` here but to `data` below. Please either use `data` here or use `message` everywhere below.